src/lib/kdf/sp800_56a/sp800_56c_one_step.h

```diff
@@ -98,12 +100,73 @@ class SP800_56A_HMAC final : public KDF {
       /**
       * @param mac the HMAC to use as the auxiliary function
       */
-      explicit SP800_56A_HMAC(std::unique_ptr<MessageAuthenticationCode> mac);
+      explicit SP800_56C_One_Step_HMAC(std::unique_ptr<MessageAuthenticationCode> mac);
 
    private:
       std::unique_ptr<MessageAuthenticationCode> m_mac;
 };
 
+/**
+ * NIST SP800-56Cr2 One-Step KDF using KMAC (Abstract class)
+ */
+class SP800_56A_One_Step_KMAC_Abstract : public KDF {
+   public:
+      /**
+      * Derive a key using the SP800-56A KDF.
+      *
+      * The implementation hard codes the context value for the
+      * expansion step to the empty string.
+      *
+      * @param key derived keying material K_M
+      * @param key_len the desired output length in bytes
+      * @param secret shared secret Z
+      * @param secret_len size of Z in bytes
+      * @param salt ignored
+      * @param salt_len ignored
+      * @param label label for the expansion step
+      * @param label_len size of label in bytes
+      *
+      * @throws Invalid_Argument key_len > 2^32 or MAC is not a HMAC
```

Comment:
Incorrect comment re HMAC

Reply:
This does not even check if MAC is an HMAC. This check is performed in the constructor. The max key length is also wrong 😅